### PR TITLE
Basic chunking using limit of 200 and iterating until all posts have

### PR DESCRIPTION
**Pull request by willdoran**
This pull request makes the following changes:
- Added basic chunking
- On completion of first request, we calculate how many additional request we should make for the given limit and if we need an additional request for the last set of posts if remainder is less than limit
- Changed to automatically add more posts
- Separate map clearing code

Test these changes by:
-

Fixes ushahidi/platform# .

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/647)

### CODE REVIEW

File: app/main/posts/views/post-view-map.directive.js [hunk 1]
```diff
@@ -85,6 +88,7 @@ function PostViewMap(PostEndpoint, Maps, _, PostFilters, L, $q, $rootScope, $com
                 return $scope.filters;
             }, function (newValue, oldValue) {
                 if (newValue !== oldValue) {
+                    clearData();
                     reloadMapPosts();
```
Comment (rjmackay) [inline app/main/posts/views/post-view-map.directive.js]: We need to make sure we show some kind of `loading` status in the UI while this happens now.

File: app/main/posts/views/post-view-map.directive.js [hunk 2]
```diff
@@ -94,12 +98,35 @@ function PostViewMap(PostEndpoint, Maps, _, PostFilters, L, $q, $rootScope, $com
             loadPosts().then(addPostsToMap);
         }
 
-        function loadPosts(query) {
+        function loadPosts(query, offset) {
             query = query || PostFilters.getQueryParams($scope.filters);
-
+            query.has_location = 'mapped';
+            offset = offset || 0;
+            var limit = 200,
```
Comment (rjmackay) [inline app/main/posts/views/post-view-map.directive.js]: Could we hoist this to the top of the directive? ie. so its easy to find and tweak the magic number later when we need to.

File: app/main/posts/views/post-view-map.directive.js [hunk 3]
```diff
@@ -94,12 +98,35 @@ function PostViewMap(PostEndpoint, Maps, _, PostFilters, L, $q, $rootScope, $com
...
+                    var remainder = (posts.total - limit) % limit;
+
+                    while (evenChunks > 0) {
+                        evenChunks -= 1;
+                        offset += limit;
+                        loadPosts(query, offset).then(addPostsToMap);
+                    }
+
+                    if (remainder > 0) {
+                        offset += remainder;
```
Comment (rjmackay) [inline app/main/posts/views/post-view-map.directive.js]: There's no need to use an exact limit. You can just add 1 to the number of chunks if there is any remainder

File: app/main/posts/views/post-view-map.directive.js [hunk 4]
```diff
@@ -94,12 +98,35 @@ function PostViewMap(PostEndpoint, Maps, _, PostFilters, L, $q, $rootScope, $com
...
+                offset: offset
+            });
             $scope.isLoading = true;
-            return PostEndpoint.geojson(query).$promise.then(function (posts) {
+            return PostEndpoint.geojson(conditions).$promise.then(function (posts) {
                 $scope.isLoading = false;
+
+                // Retrieve remaining posts
+                if (offset === 0 && posts.total > limit) {
+                    var evenChunks = Math.floor((posts.total - limit) / limit);
```
Comment (rjmackay) [inline app/main/posts/views/post-view-map.directive.js]: can we rename this to something clearer? ie. numberOfChunks?

Comment (willdoran) [inline app/main/posts/views/post-view-map.directive.js]: 👍

File: app/main/posts/views/post-view-map.directive.js (hunk 3 above)
Comment (willdoran) [inline app/main/posts/views/post-view-map.directive.js]: 👍

**Pull request merged**